### DISCUSSION TITLE
Redirect subdomain to a third-party website

### DISCUSSION THREAD

**Discussion by DavidSalomonDev** [vercel/community#230] [Help] [answered]
Hi, I have both domains: `david-salomon.com` and `davidsalomon.dev` and I have a blog on Hashnode with `blog.davidsalomon.dev` and I want to make `blog.david-salomon.com` redirect to my Hashnode blog.

However, I've tried several solutions and I can't make `blog.david-salomon.com` redirecting to `blog.davidsalomon.dev` here's what I've tried.

1. Creating an alias in david-salomon.com in DNS UI
For some reason it doesn't work

2. Create a vercel.json with the value 
```
"redirects": [
      { "source": "blog.david-salomon.com", "destination": "https://blog.davidsalomon.dev" }
    ]
```

3. Create a redirection from the 'Redirect' menu as per https://vercel.com/changelog/domains-can-now-be-redirected-with-a-custom-status-code 
The 'Edit' option or the 'Domains' option are not available 
![image](https://user-images.githubusercontent.com/57274941/112768134-adb51b00-8fd7-11eb-8e38-98912fd74936.png)

Any help will be appreciated.

**Comment by fu-sen** [comment, marked as answer]
This is probably what you're expecting in `vercel.json`: 
```
{
  "redirects": [
   { "source": "/(.*)", "destination": "https://blog.davidsalomon.dev/$1", "permanent": true }
  ]
}
```
When using `redirects`, ▲Vercel does not provide `301`. You can use `308` instead.

The lack of the Domains option is another issue.
You are looking at your account page. Not there: https://vercel.com/account

Did you assign the domain to the project? 
If you're doing `git push` or` vercel --prod`, you should have created that project.
see the project page. There is **View Domains** there: https://vercel.com/dashboard/projects

**Comment by fu-sen** [comment]
Note: I'm using Windows, so it's `nslookup`:

```
PS C:\Users\balloon> nslookup david-salomon.com
Non authoritative answer:
Name:    david-salomon.com
Address:  76.76.21.21

PS C:\Users\balloon> nslookup blog.david-salomon.com
Non authoritative answer:
Name:    cname.vercel-dns.com
Address:  76.76.21.21
Aliases:  blog.david-salomon.com

PS C:\Users\balloon> ▄
```

These are correctly directed to ▲Vercel. The name server has a cache, so you had to wait for the server reference to change.

**Comment by brianshano** [comment]
Did you solve this?
I was trying to do similar but from the vercel configuration level. 
Following this https://vercel.com/support/articles/pointing-subdomains-to-external-services

When I add the subdomain with the CNAME value to redirect to, I get this error message:
"Your Domain isn't using Vercel nameservers. In order for the changes to your DNS Records to get applied, however, it needs to. [Learn More](https://vercel.link/configure-vercel-nameservers)"

**Reply by fu-sen** [reply]
This problem occurs not only in Vercel but also in the name server settings of the domain. 
The name server has a cache called TTL. Therefore, please note that even if you set it, it will not refer to the new server immediately.
After changing the server, it may take hours to days for it to take effect.
Make sure the domain you set refers to the name server, as shown above. for example:
https://toolbox.googleapps.com/apps/dig/
